server/controllers/auth/signup.js

```diff
@@ -1,3 +1,39 @@
-exports.signup = (req, res) => {
-  res.send(' Hello from signup file');
+const { hash } = require('bcrypt');
+const { jwtSign } = require('../../helpers');
+const { signupSchema } = require('./validation/signup-validation');
+const { users: { getUserByEmailOrUsername, addUser, addUserInterests } } = require('../../database/queries');
+
+
+exports.signup = (req, res, next) => {
+  const key = process.env.KEY;
+  const {
+    username, email, password, nativeLangId, learningLangId, interestsId,
+  } = req.body;
+  signupSchema.validate({
+    username, email, password, nativeLangId, learningLangId, interestsId,
```

Comment:
you using this object twice 
you can write it like this ---> it's will be more readable and better 
```js 
  const userIfo = {
    username, email, password, nativeLangId, learningLangId, interestsId,
  };

```

Reply:
even better, you might just pass req.body directly. If you notice, you destructure from req.body, then take everything and create a new object from it..